Core message consumer fails and blocks the scenario

### What happened?

Core logs:

```
│ [13:23:29,867] [INFO]           [InfraManager] successfully cached http://osrd-rec-editoast.osd-rec01.svc.cluster.local/infra/62/railjson/         │
│ [13:23:29,867] [INFO]         [WorkerCommand] monitoring type: datadog                                                                            │
│ [13:23:30,340] [DEBUG]   [ConsumerWorkService] Creating executor service with 4 thread(s) for consumer work service                                │
│ [13:23:30,642] [INFO]         [WorkerCommand] consume ended                                                                                       │
│ [13:23:30,657] [INFO]         [WorkerCommand] received request for path /infra_load                                                               │
│ [13:23:30,707] [INFO]         [WorkerCommand] request for path /infra_load processed                                                              │
│ [13:23:30,709] [INFO]         [WorkerCommand] received request for path /v2/pathfinding/blocks                                                    │
│ [13:23:31,264] [INFO]         [WorkerCommand] request for path /v2/pathfinding/blocks processed                                                   │
│ [13:23:31,411] [INFO]         [WorkerCommand] received request for path /v2/path_properties                                                       │
│ [13:23:31,494] [INFO]         [WorkerCommand] request for path /v2/path_properties processed                                                      │
│ [13:23:47,936] [INFO]         [WorkerCommand] received request for path /infra_load                                                               │
│ [13:23:47,942] [INFO]         [WorkerCommand] request for path /infra_load processed                                                              │
│ [13:24:12,087] [INFO]         [WorkerCommand] received request for path /v2/pathfinding/blocks                                                    │
│ [13:24:12,177] [INFO]         [WorkerCommand] request for path /v2/pathfinding/blocks processed                                                   │
│ [13:24:12,343] [INFO]         [WorkerCommand] received request for path /v2/path_properties                                                       │
│ [13:24:12,355] [INFO]         [WorkerCommand] request for path /v2/path_properties processed                                                      │
│ [13:29:05,047] [ERROR]         [WorkerCommand] consumer cancelled     
has context menu
```

### What did you expect to happen?

_No response_

### How can we reproduce it (as minimally and precisely as possible)?

Sadly, I cannot easily reproduce the bug.

### On which environments the bug occurs?

Recette (SNCF)

### On which browser the bug occurs?

Firefox

### OSRD version (top right corner `Account` button > `Informations`)

80973c5

--- a/core/gradle/libs.versions.toml
+++ b/core/gradle/libs.versions.toml
@@ -67,7 +67,7 @@ opentelemetry-instrumentation-annotations = { module = 'io.opentelemetry.instrum
 
 kaml = { module = 'com.charleskorn.kaml:kaml', version = '0.66.0' } # Apache 2.0
 
-amqp-client = { module = 'com.rabbitmq:amqp-client', version = '5.23.0' }
+amqp-client = { module = 'com.rabbitmq:amqp-client', version = '5.24.0' }
 
 [plugins]
 # kotlin

--- a/core/src/main/java/fr/sncf/osrd/App.java
+++ b/core/src/main/java/fr/sncf/osrd/App.java
@@ -41,7 +41,12 @@ public class App {
         }
 
         // run the user command
-        var statusCode = commands.get(commandName).run();
+        var statusCode = 1;
+        try {
+            statusCode = commands.get(commandName).run();
+        } catch (Exception e) {
+            e.printStackTrace(System.err);
+        }
         System.exit(statusCode);
     }
 }

--- a/core/src/main/java/fr/sncf/osrd/cli/WorkerCommand.kt
+++ b/core/src/main/java/fr/sncf/osrd/cli/WorkerCommand.kt
@@ -293,6 +293,17 @@ class WorkerCommand : CliCommand {
                         executionTimeMS / 1_000.0
                     )
                 }
+
+            val terminatorCallback =
+                fun(message: Delivery) {
+                    try {
+                        callback(message)
+                    } catch (t: Throwable) {
+                        t.printStackTrace(System.err)
+                        exitProcess(1)
+                    }
+                }
+
             channel.basicConsume(
                 WORKER_REQUESTS_QUEUE,
                 false,
@@ -302,9 +313,9 @@ class WorkerCommand : CliCommand {
                         // We directly process the message with no dispatch if there's too many
                         // locally queued tasks. Prevents the worker from consuming all the rabbitmq
                         // at once, which would mess with the stats and automatic scaling.
-                        callback(message)
+                        terminatorCallback(message)
                     } else {
-                        executor.execute { callback(message) }
+                        executor.execute { terminatorCallback(message) }
                     }
                 },
                 { _ ->

--- a/core/src/main/java/fr/sncf/osrd/cli/WorkerCommand.kt
+++ b/core/src/main/java/fr/sncf/osrd/cli/WorkerCommand.kt
@@ -3,7 +3,10 @@ package fr.sncf.osrd.cli
 import com.beust.jcommander.Parameter
 import com.beust.jcommander.Parameters
 import com.rabbitmq.client.*
-import fr.sncf.osrd.api.*
+import fr.sncf.osrd.api.ElectricalProfileSetManager
+import fr.sncf.osrd.api.InfraLoadEndpoint
+import fr.sncf.osrd.api.InfraManager
+import fr.sncf.osrd.api.VersionEndpoint
 import fr.sncf.osrd.api.api_v2.conflicts.ConflictDetectionEndpointV2
 import fr.sncf.osrd.api.api_v2.path_properties.PathPropEndpoint
 import fr.sncf.osrd.api.api_v2.pathfinding.PathfindingBlocksEndpointV2
@@ -20,6 +23,7 @@ import java.io.InputStream
 import java.util.concurrent.LinkedBlockingQueue
 import java.util.concurrent.ThreadPoolExecutor
 import java.util.concurrent.TimeUnit
+import kotlin.system.exitProcess
 import okhttp3.OkHttpClient
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
@@ -303,7 +307,10 @@ class WorkerCommand : CliCommand {
                         executor.execute { callback(message) }
                     }
                 },
-                { _ -> logger.error("consumer cancelled") },
+                { _ ->
+                    logger.error("consumer cancelled")
+                    exitProcess(0)
+                },
                 { consumerTag, e ->
                     logger.info("consume shutdown: {}, {}", consumerTag, e.toString())
                 }

